mirgecom/integrators/lsrk.py

```diff
@@ -0,0 +1,157 @@
+"""Timestepping routines for low-storage variants of explicit Runge-Kutta methods.
+
+.. autofunction:: lsrkeuler_step
+.. autofunction:: lsrk54_step
+.. autofunction:: lsrk144_step
+"""
+
+__copyright__ = """
+Copyright (C) 2020 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+from dataclasses import dataclass
+
+import numpy as np
+
+
+__all__ = ("lsrkeuler_step", "lsrk54_step", "lsrk144_step")
+
+
+@dataclass(frozen=True)
+class LSRKCoefficients:
+    """Dataclass which defines a given low-storage Runge-Kutta (LSRK) scheme.
+
+    The methods are determined by the provided `A`, `B` and `C` coefficient arrays.
+    """
+
+    A: np.ndarray
+    B: np.ndarray
+    C: np.ndarray
+
+
+def lsrk_step(coefs, state, t, dt, rhs):
+    """Take one step using a low-storage Runge-Kutta method."""
+    k = 0.0 * state
+    for i in range(len(coefs.A)):
+        k = coefs.A[i]*k + dt*rhs(t + coefs.C[i]*dt, state)
+        state += coefs.B[i]*k
+
+    return state
+
+
+LSRKEulerCoefs = LSRKCoefficients(
+    A=np.array([0.]),
+    B=np.array([1.]),
+    C=np.array([0.]))
+
+
+def lsrkeuler_step(state, t, dt, rhs):
```

Comment:
Low-storage Euler, eh? :)

I get the consistency angle here, but it reads funny to me. (vs just `euler`) What do you think?

Reply:
It is a bit silly looking at it now lol. I just wanted to keep it consistent (`lsrkX_step`). But I am totally happy with keeping it as before (`euler_step`). Though in all honestly, we should NOT be using explicit Euler in production ;)